x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,402 @@
+ * if the customer is using Native Connectors, agentless is available, but the integration server is missing, they are told that Integrations Server must be added
+ * if the customer is using Native Connectors, and agentless is unavailable, they are told that they must convert their connectors to Connector Clients
+ * if the customer was using "native" connectors that don't match our connector service types, they must delete them or convert them to connector clients.
+ */
+export async function getNativeConnectorDeprecations(
+  ctx: GetDeprecationsContext,
+  hasAgentless: boolean,
+  hasFleetServer: boolean,
+  cloud: CloudSetup,
+  docsUrl: string
+): Promise<DeprecationsDetails[]> {
+  const client = ctx.esClient.asInternalUser;
+  const connectors: Connector[] = await fetchConnectors(client, undefined, false, undefined);
+  const nativeConnectors = connectors.filter((hit) => hit.is_native);
+  if (nativeConnectors.length === 0) {
+    return []; // no deprecations to register if there are no Native Connectors
+  } else {
+    const deprecations: DeprecationsDetails[] = [];
+
+    if (nativeConnectors.length > 0 && !hasAgentless) {
+      // you can't have elastic-managed connectors in an environment that Elastic doesn't manage
+      // ... and agentless is available in every Elastic-managed environment
+
+      deprecations.push({
+        level: 'critical',
+        deprecationType: 'feature',
+        title: i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.title', {
+          defaultMessage:
+            'Connectors with `is_native: true` are not supported in self-managed environments',
+        }),
+        message: {
+          type: 'markdown',
+          content: i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.message', {
+            defaultMessage:
+              '"Native Connectors" are managed services in Elastic-managed environments such as Elastic Cloud Hosted and ' +
+              'Elastic Serverless. Any connectors with `is_native: true` must be converted to connector clients or deleted ' +
+              'before this upgrade can proceed.',
+          }),
+        },
+        documentationUrl: docsUrl,
+        correctiveActions: {
+          manualSteps: [
+            i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.listConnectors', {
+              defaultMessage: 'Enumerate all connector records',
+            }),
+            i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.convertPretenders', {
+              defaultMessage:
+                'Select "Convert to Client" for any connectors where `is_native: true`.',
+            }),
+          ],
+          api: {
+            method: 'POST',
+            path: '/internal/enterprise_search/deprecations/convert_connectors_to_client',
+            body: {
+              ids: nativeConnectors.map((it) => it.id),
+            },
+          },
+        },
+      });
+    } else {
+      const nativeServiceTypes = [
```

Comment:
nit: because this is a long list, maybe it would be easier to print the much shorter list of unsupported service types and say 'if you have one of these with `is_native=true` you'll need to convert".

Reply:
What this is actually trying to do is cover customer-specific service types. Like if we look at telemetry, we find a bunch of service types that are not ours. 

<img width="641" alt="Screenshot 2025-01-09 at 1 32 38 PM" src="https://github.com/user-attachments/assets/9b908e8d-066a-4f76-9f9d-d2dd2911c3ff" />

Reply:
ah yes! this dawned on me later in the day, that there would be custom connectors at play here too 👍